src/flexmeasures_client/response_handling.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import asyncio
+from typing import TYPE_CHECKING
+
+from aiohttp import ContentTypeError
+
+from flexmeasures_client.constants import CONTENT_TYPE
+
+if TYPE_CHECKING:  # Only imports the below statements during type checking
+    from flexmeasures_client.client import FlexMeasuresClient
+
+
+async def check_response(self: FlexMeasuresClient, response):
+    """
+    <300: passes
+    401: reauthenticate
+    todo: 503 + Retry-After header: poll again
```

Comment:
In the `request` docstring you mention 408 explicitly. I think we should check for that here explicitly. Note that in the one test we have in which the server responds with a 408, the client actually gives up before the server sends something back.

Reply:
-> New issue (should include test)